testing/tests/DevExpress.ui.widgets.treeList/editing.tests.js

```diff
@@ -286,6 +286,62 @@ QUnit.test('AddRow method should expand row and add item after parent', function
     assert.strictEqual(rows[2].node.parent.key, 1, 'row 2 node parent');
 });
 
+QUnit.test('AddRow method should return Deferred with collapsed parent', function(assert) {
+    // arrange
+    this.options.editing.allowAdding = true;
+
+    this.setupTreeList();
+    this.rowsView.render($('#treeList'));
+    this.clock.tick();
+
+    // assert
+    assert.equal(this.getVisibleRows().length, 1, 'one visible row');
+
+    // act
+    this.addRow(1).done(() => {
+        // assert
+        assert.ok(true, 'addRow returns Deferred');
+        assert.equal(this.getVisibleRows().length, 3, 'parent is expanded and one more row is added');
+    });
+
+    this.clock.tick();
+});
+
+QUnit.test('Sequential adding of a row after adding the previous using Deferred (T844118)', function(assert) {
+    // arrange
+    const initNewRowCalls = [];
+
+    this.options.editing.allowAdding = true;
+    this.options.onInitNewRow = (e) => {
+        initNewRowCalls.push(e.data.parentId);
+    };
+
+    this.setupTreeList();
+    this.rowsView.render($('#treeList'));
+    this.clock.tick();
+
+    // assert
+    assert.equal(this.getVisibleRows().length, 1, '1 visible row');
+
+    // act
+    const addRowAfterDeferredResolve = (sequence, index) => {
```

Comment:
Rename sequence to parentIds because 'sequence' name does not say anything what it is about.

Reply:
Done